dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -132,23 +144,58 @@ public void setCurrentTree(Tree currentTree) {
     /**
      * Construct an object that can perform a org.checkerframework.dataflow analysis over a control
      * flow graph. The transfer function is set later using {@code setTransferFunction}.
+     *
+     * @deprecated as {@code env} is moved to {@link
+     *     org.checkerframework.framework.flow.CFAbstractAnalysis}, this helper constructor will be
+     *     removed in next major update. Use {@link #Analysis()}, {@link #Analysis(TransferFunction,
+     *     int)}, {@link #Analysis(TransferFunction)} or {@link #Analysis(int)} instead
+     * @param env associated processing environment
      */
+    @Deprecated
     public Analysis(ProcessingEnvironment env) {
         this(null, -1, env);
     }
 
+    /**
+     * Construct an object that can perform a org.checkerframework.dataflow analysis over a control
+     * flow graph. The transfer function is set later using {@code setTransferFunction}.
```

Comment:
@wmdietl This comment was copied from the old method. `setTransferFunction` was removed in c6e8ae771fea7064fc4a8a669cb2210c3e1f52c4.  Do you know how the transfer function should be set later?

Reply:
@smillst I just talked about this with Prof. Dietl, `setTransferFunction` should be removed from the comments. `transferfunction` can be set by the subclasses later. For example, in `CFAbstractAnalysis`, it is set by [createTransferFunction()](https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/flow/CFAbstractAnalysis.java#L72).

Thanks a lot for your reviews!